cli/src/semgrep/error_handler.py

```diff
@@ -72,13 +73,17 @@ def send(self, exit_code: int) -> int:
             f"Sending to fail-open endpoint {state.env.fail_open_url} since fail-open is configured to {self.suppress_errors}"
         )
 
-        headers = self.payload.get("headers", {})
-        headers["User-Agent"] = str(state.app_session.user_agent)
-        self.payload["headers"] = headers
+        token = auth.get_token()
+        headers = {
+            "User-Agent": str(state.app_session.user_agent),
+            "Authorization": f"Bearer {token or ''}",
```

Comment:
will this token be captured even if we fail early in the process?

Reply:
Yes. The token is pulled from an env variable (most commonly for CI) or a `.semgrep/settings.yml` file. So regardless of the failure, the token should be available (or worst case just returns `None`)